### ISSUE TITLE
calcite-input: ensure that the cross icon that appears at the end of the input field when adding the clearable attribute is keyboard operable

### ISSUE THREAD

**Issue by sakship2204** [Esri/calcite-design-system#9718]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

When we add the clearable attribute to the calcite-input component, we get a cross icon to clear the text that we enter in the field. This clearable icon is not keyboard operable. We want to make it accessible using tab key for the specially abled users.

![Screenshot 2024-07-01 at 11 38 39 AM](https://github.com/Esri/calcite-design-system/assets/148040911/789e6787-17da-4e66-9b99-80443b2ae914)

### Actual Behavior

The clear button at the end of the calcite-input is not keyboard operable.

### Expected Behavior

The clear button should be keyboard operable.

### Reproduction Sample

--

### Reproduction Steps

1. Enter any text in the calcite-input component
2. Press Tab repeatedly to navigate through the focusable elements on the page.
3. Notice that the links are never focused.
![image](https://github.com/Esri/calcite-design-system/assets/148040911/f41767fd-7b0e-4b89-a269-55bc93f71a0c)

### Reproduction Version

--

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise

**Comment by macandcheese** [comment]
This is an intentional decision. Discussion: https://github.com/Esri/calcite-design-system/issues/2381

**Comment by geospatialem** [comment]
> This is an intentional decision. Discussion: #2381

Since the component is clearable via the `esc` key when `clearable` is set on the component, the button does not need to be keyboard accessible. Closing the above out as out of scope.

We will however add keyboard navigation with the `esc` key to showcase when `clearable` is present the value will be cleared from the component.
